agent/lib/kontena/workers/node_stats_worker.rb

```diff
@@ -0,0 +1,300 @@
+require 'net/http'
+require 'vmstat'
+
+require_relative '../models/node'
+require_relative '../helpers/rpc_helper'
+require_relative '../helpers/stats_helper'
+
+module Kontena::Workers
+  class NodeStatsWorker
+    include Celluloid
+    include Celluloid::Notifications
+    include Kontena::Logging
+    include Kontena::Observer
+    include Kontena::Helpers::RpcHelper
+    include Kontena::Helpers::StatsHelper
+
+    attr_reader :statsd, :stats_since
+
+    PUBLISH_INTERVAL = 60
+
+    # @param [Boolean] autostart
+    def initialize(autostart = true)
+      @node = nil
+      @statsd = nil
+      @stats_since = Time.now
+      @container_seconds = 0
+      @previous_cpu = Vmstat.cpu
+      @previous_network = Vmstat.network_interfaces
+      subscribe('container:event', :on_container_event)
+      info 'initialized'
+      async.start if autostart
+    end
+
+    def start
+      observe(Actor[:node_info_worker]) do |node|
+        configure_statsd(node)
+        @node = node
+      end
+
+      start_publish_loop
+    end
+
+    def start_publish_loop
+      loop do
+        sleep PUBLISH_INTERVAL
+        self.publish_node_stats if rpc_client.connected?
```

Comment:
The `if rpc_client.connected?` check is new? This also stops statsd metrics if the RPC is down.